R/modify-records.R

```diff
@@ -323,6 +323,131 @@ get_table_columns <- function(.mod, from_data = TRUE){
   return(table_cols)
 }
 
+
+# Function to translate NONMEM operators to R operators
+translate_operator <- function(expr) {
+  expr <- gsub("\\.EQ\\.", "==", expr)
+  expr <- gsub("\\.NE\\.", "!=", expr)
+  expr <- gsub("\\.LT\\.", "<", expr)
+  expr <- gsub("\\.LE\\.", "<=", expr)
+  expr <- gsub("\\.GT\\.", ">", expr)
+  expr <- gsub("\\.GE\\.", ">=", expr)
+  return(expr)
+}
+
+# Function to invert an expression (for ignore statements)
+invert_expression <- function(expr) {
+  expr <- dplyr::case_when(
+    grepl("==", expr) ~ gsub("==", "!=", expr),
+    grepl("!=", expr) ~ gsub("!=", "==", expr),
+    grepl("<=", expr) ~ gsub("<=", ">=", expr),
+    grepl(">=", expr) ~ gsub(">=", "<=", expr),
+    grepl("<", expr) ~ gsub("<", ">", expr),
+    grepl(">", expr) ~ gsub(">", "<", expr),
+    TRUE ~ expr
+  )
+  return(expr)
+}
+
+
+#' Filter `NONMEM` input data based on `IGNORE` and `ACCEPT` record options
+#'
+#' @param .mod a `bbi_nonmem_model` object
+#' @param data a starting dataset
+#' @keywords internal
+filter_nm_data <- function(.mod, data = nm_data(.mod)){
```

Comment:
Suggest to break this function down a bit more. Especially the code that processes `$DATA` block. If you isolate that into its own function, it will be _much_ easier to test.

Reply:
Just seeing this but coincidentally did just that